### ISSUE TITLE
BottomNavigationView does not reappear

### ISSUE THREAD

**Issue by shashank-0-0** [kiwix/kiwix-android#3407]
**Describe the bug**

There is a scenario where the bottom navigation view completely disappears / never re enter's again / not available for the user.

**Expected behavior**

The bottom navigation view shouldn't disappear and should be available to the user for navigation always.

**Steps to reproduce the behavior:**
1. Navigate to library screen
2. Have at most items/books where the scrolling behaviour of the bottom view is just enabled (5 items in my case).
3. Delete an item when the bottom navigation view is scrolled away
4. Bottom navigation view is lost ;( , hence user cannot navigate.

**Reason**
If we delete an item when the bottom view is scrolled away and the scrolling of the recycler view is disabled/stopped due to less no of items ,the bottom view cannot come back / re enter

**Screen Recording**

https://github.com/kiwix/kiwix-android/assets/111702455/1e378d17-c750-4617-ac25-2de43522f2ad

**Environment**
- Version of Kiwix Android : 3.7.1 develop version
- Device : Realme 3i
- OS version : 10

**Comment by shashank-0-0** [comment]
@kelson42 @MohitMaliFtechiz I am excited to make my 3rd pr :) let me fix it .

**Comment by kelson42** [comment]
@gouri-panda @MohitMaliFtechiz You confirm the bug, sounds surprising considering that last time we had a bug I had insisted to test this part accurately.

**Comment by gouri-panda** [comment]
@kelson42 @shashank-0-0 I can't reproduce this issue.  @shashank-0-0 can you give me the logs if possible APk is a good idea too. Thanks!

**Comment by shashank-0-0** [comment]
@gouri-panda Here is the apk file 
https://www.dropbox.com/s/lhar47cmt9azi68/kiwix-arm64-v8a-debug.apk?dl=0

I also reproduced it in 
OS : Android 9
Device : vivo 

More info : I had 5 items /books in both the devices and then i deleted one when the bottom view is scrolled away.

**Comment by gouri-panda** [comment]
@shashank-0-0 Thanks!
>Solution:
we have to bring the bottom view back when user decides to delete an item i.e. when delete action is detected that way we won't loose it.

Looks like a good solution :)

**Comment by MohitMaliFtechiz** [comment]
@shashank-0-0, Thank you for discovering the bug. I can confirm that we have implemented a check for scrolling up. However, we encountered an issue when deleting an item. Specifically, when there are 5 items in the recyclerview, deleting an item causes the recyclerview's height to change. As a result, when scrolling up, the scrolling toolbar initially scrolls down instead. This is why the provided code snippet does not work for this particular scenario.

https://github.com/kiwix/kiwix-android/blob/0d631d9d8f8d28000ea4c0ad35c88d03e6941174/app/src/main/java/org/kiwix/kiwixmobile/nav/destination/library/LocalLibraryFragment.kt#L223

I appreciate the solution you suggested and support their implementation.
